src/server_manager/web_app/app.ts

```diff
@@ -864,42 +885,90 @@ export class App {
+  private async setDefaultDataLimit(limit: server.DataLimit) {
     if (!limit) {
       return;
     }
-    const previousLimit = this.selectedServer.getAccessKeyDataLimit();
+    const previousLimit = this.selectedServer.getDefaultDataLimit();
     if (previousLimit && limit.bytes === previousLimit.bytes) {
       return;
     }
     const serverView = this.appRoot.getServerView(this.appRoot.selectedServerId);
     try {
-      await this.selectedServer.setAccessKeyDataLimit(limit);
+      await this.selectedServer.setDefaultDataLimit(limit);
       this.appRoot.showNotification(this.appRoot.localize('saved'));
-      serverView.accessKeyDataLimit = dataLimitToDisplayDataAmount(limit);
+      serverView.defaultDataLimit = dataLimitToDisplayDataAmount(limit);
       this.refreshTransferStats(this.selectedServer, serverView);
       // Don't display the feature collection disclaimer anymore.
       serverView.showFeatureMetricsDisclaimer = false;
       window.localStorage.setItem('dataLimits-feature-collection-notification', 'true');
     } catch (error) {
-      console.error(`Failed to set access key data limit: ${error}`);
+      console.error(`Failed to set server default data limit: ${error}`);
       this.appRoot.showError(this.appRoot.localize('error-set-data-limit'));
-      serverView.accessKeyDataLimit = dataLimitToDisplayDataAmount(
-          previousLimit || await computeDefaultAccessKeyDataLimit(this.selectedServer));
-      serverView.isAccessKeyDataLimitEnabled = !!previousLimit;
+      serverView.defaultDataLimit = dataLimitToDisplayDataAmount(
+          previousLimit || await computeDefaultDataLimit(this.selectedServer));
+      serverView.isDefaultDataLimitEnabled = !!previousLimit;
     }
   }
 
-  private async removeAccessKeyDataLimit() {
+  private async removeDefaultDataLimit() {
     const serverView = this.appRoot.getServerView(this.appRoot.selectedServerId);
     try {
-      await this.selectedServer.removeAccessKeyDataLimit();
+      await this.selectedServer.removeDefaultDataLimit();
       this.appRoot.showNotification(this.appRoot.localize('saved'));
       this.refreshTransferStats(this.selectedServer, serverView);
     } catch (error) {
-      console.error(`Failed to remove access key data limit: ${error}`);
+      console.error(`Failed to remove server default data limit: ${error}`);
       this.appRoot.showError(this.appRoot.localize('error-remove-data-limit'));
-      serverView.isAccessKeyDataLimitEnabled = true;
+      serverView.isDefaultDataLimitEnabled = true;
+    }
+  }
+
+  private maybeAdminKey(key: DisplayAccessKey, view: ServerView) {
+    return key.id === MY_CONNECTION_USER_ID ? view.myConnection : key;
+  }
+
+  private findModifiedKey(id: server.AccessKeyId, view: ServerView) {
+    return this.maybeAdminKey(view.accessKeyRows.find(key => key.id === id), view);
+  }
+
+  private async savePerKeyDataLimit(ui: OutlinePerKeyDataLimitDialog) {
```

Comment:
Please rename `ui` to `dialog`.

Reply:
Done in 82225e38973141a96e7c7f74f9a0cc249e5c2f1f